Gui/opensim/view/src/org/opensim/view/experimentaldata/MotionObjectPointForce.java

```diff
@@ -180,7 +180,7 @@ public void generateDecorations(boolean fixed, ModelDisplayHints hints, State st
             Transform xform = new Transform();
             
             xform.setP(new Vec3(point[0], point[1], point[2]));
-            appendToThis.push_back(new DecorativeArrow(new Vec3(0, 0, 0)).setBodyId(0).setColor(new Vec3(0., 1., .0)).setOpacity(0.5).setIndexOnBody(getStartIndexInFileNotIncludingTime()).setTransform(xform));            
+            appendToThis.push_back(new DecorativeArrow(new Vec3(0, 0, 0)).setBodyId(0).setColor(color).setOpacity(0.5).setIndexOnBody(getStartIndexInFileNotIncludingTime()).setTransform(xform));            
```

Comment:
This line of code is indecipherable. Please please describe what this is supposed to be doing? `setBodyId(0)...setIndexOnBody(getStartIndexInFileNotIncludingTime())` is very confusing. What is `getStartIndexInFileNotIncludingTime()` and why is it relevant or necessary in this context?

Reply:
@aseth1 The cascading calls are used all the time to set attributes on Decorative Geometry as was done here https://github.com/opensim-org/opensim-core/blob/dc81fe58bccc3061ff26e09806ba085681a71d2d/OpenSim/Sandbox/futureMarkerPreview.cpp#L75 it was built into this API by Sherm for this reason.

getStartIndexInFileNotIncludingTime() returns the index of the data visualized by the "Arrow" in the corresponding Storage file, it gets computed and cached on construction. IndexOnBody is used for book-keeping to generate a unique name to pass to the visualizer (since these are not components with full pathnames etc.)

Reply:
There is no issue with the cascading functionality. The issues is with documenting the code for others to understand what it is doing. Please comment your code.